### PR TITLE
Expose BatteryPool and PowerDistributingHandle from `sdk.microgrid`

### PR DESCRIPTION
**Pull request by sahas-subramanian-frequenz**

### CODE REVIEW

File: src/frequenz/sdk/microgrid/_data_pipeline.py [hunk 1]
```diff
@@ -309,6 +309,31 @@ def ev_charger_pool(ev_charger_ids: set[int] | None = None) -> EVChargerPool:
...
 
+def battery_pool(battery_ids: set[int] | None = None) -> BatteryPool:
+    """Return the corresponding BatteryPool instance for the given ids.
+
+    If a BatteryPool instance for the given ids doesn't exist, a new one is
+    created and returned.
+
+    Args:
+        battery_ids: Optional set of IDs of batteries to be managed by the
+            BatteryPool.
```
Comment (mathias-baumann-frequenz) [inline src/frequenz/sdk/microgrid/_data_pipeline.py]: The documentation could mention what is returned when no ids are given

Comment (sahas-subramanian-frequenz) [reply src/frequenz/sdk/microgrid/_data_pipeline.py]: true,  added an extra commit with docs.

File: src/frequenz/sdk/microgrid/_data_pipeline.py [hunk 2]
```diff
@@ -300,15 +300,42 @@ def ev_charger_pool(ev_charger_ids: set[int] | None = None) -> EVChargerPool:
...
 
+def battery_pool(battery_ids: abc.Set[int] | None = None) -> BatteryPool:
+    """Return the corresponding BatteryPool instance for the given ids.
+
+    If a BatteryPool instance for the given ids doesn't exist, a new one is
+    created and returned.
+
+    Args:
+        battery_ids: Optional set of IDs of batteries to be managed by the
+            BatteryPool.  If not specified, all batteries available in the
```
Comment (mathias-baumann-frequenz) [inline src/frequenz/sdk/microgrid/_data_pipeline.py:321]: Do we want to use double space after a `.`? Is that a policy I missed? Same above

Comment (sahas-subramanian-frequenz) [reply src/frequenz/sdk/microgrid/_data_pipeline.py:321]: We don't have a policy for or against it.  But either way, markdown renderers are able to keep them consistent when generating docs.

**Review by mathias-baumann-frequenz — APPROVED** [review]

**Pull request merged**